app/models/concerns/notify_concern.rb

```diff
@@ -0,0 +1,70 @@
+module NotifyConcern
```

Comment:
@xzhflying have you documented how to use your framework ? or examples ?

If I understand correctly, this concern is supposed to included by models, then they can send notifications  ?